pkg/ingester/stream.go

```diff
@@ -64,6 +66,11 @@ type chunkDesc struct {
 	lastUpdated time.Time
 }
 
+type entryWithError struct {
+	entry logproto.Entry
+	e     error
+}
```

Comment:
I think we probably want to make these pointers:

```
type entryWithError struct {
	entry *logproto.Entry
	e     *error
}
```

else when they are populated below it would be a copy I believe.

Reply:
I don't think we want to make `error` a pointer, it's already an interface.

Reply:
very true, I'm wondering if instead we should put the string here from the error

Reply:
done, reflection on error needed, so can't be string